fdbcli/GetAuditStatusCommand.actor.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * GetAuditStatusCommand.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbcli/fdbcli.actor.h"
+
+#include "fdbclient/Audit.h"
+#include "fdbclient/IClientApi.h"
+#include "flow/ThreadHelper.actor.h"
+
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+namespace {
+
+int parseLimit(StringRef input) {
+	if (input.size() > 9) {
+		return 1;
+	}
+
+	int limit = 0;
+	for (int i = input.size(); i > 0; --i) {
+		const int val = input[i - 1] - '0';
+		if (val < 0 || val > 9) {
+			return 1;
+		}
+		limit = limit * 10 + val;
+	}
+
+	return limit;
+}
+
+ACTOR Future<Void> getLatestAuditStates(Reference<IDatabase> db, AuditType type, int num) {
+	state Reference<ITransaction> tr = db->createTransaction();
+
+	loop {
+		tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+		try {
+			state ThreadFuture<RangeResult> resultFuture =
+			    tr->getRange(auditKeyRangeForType(type), num, Snapshot::False, Reverse::True);
+			RangeResult res = wait(safeThreadFutureToFuture(resultFuture));
+			for (int i = 0; i < res.size(); ++i) {
+				auto auditState = decodeAuditStorageState(res[i].value);
+				printf("AuditStorageState:\nID: %s\nType: %d\nState:%d\n",
+				       auditState.id.toString().c_str(),
+				       auditState.type,
+				       auditState.type);
+			}
+		} catch (Error& e) {
+			TraceEvent("GetLatestAuditStatesError").error(e);
+			wait(safeThreadFutureToFuture(tr->onError(e)));
+		}
+	}
+}
+
+} // namespace
+
+namespace fdb_cli {
+
+ACTOR Future<bool> getAuditStatusCommandActor(Reference<IDatabase> db, std::vector<StringRef> tokens) {
+	state AuditType type;
+
+	if (tokens.size() < 2) {
+		printUsage(tokens[0]);
+		return false;
+	}
+
+	if (tokencmp(tokens[1], "ha")) {
+		type = AuditType::ValidateHA;
+	}
+
+	int limit = 1;
+	if (tokens.size() >= 3) {
+		limit = parseLimit(tokens[2]);
```

Comment:
Instead of defining a new parsing function, can we use `parse_with_suffix()` here?